addons/udes_stock/models/stock_move.py

```diff
@@ -175,3 +175,98 @@ def split_out_move_lines(self, move_lines, **kwargs):
+        Push = self.env["stock.rule"]
+
+        done_moves = self.filtered(lambda m: m.state == "done")
+
+        # load all the move lines, grouped by location
+        move_lines_by_location = done_moves.mapped("move_line_ids").groupby("location_dest_id")
+
+        # Build mapping of push rule -> move lines to push
+        move_lines_by_push = {}
+        for location, loc_mls in move_lines_by_location:
+            # Get the push rule that moves from the location.
+            push_step = Push.get_path_from_location(location)
+            if not push_step:
+                continue
+            if push_step not in move_lines_by_push:
+                move_lines_by_push[push_step] = MoveLine.browse()
+            move_lines_by_push[push_step] |= loc_mls
+
+        created_moves = Move.browse()
+        for push, move_lines in move_lines_by_push.items():
+            created_moves |= self._create_moves_for_push(push, move_lines)
+
+        confirmed_moves = created_moves._action_confirm()
+        pickings = confirmed_moves.mapped("picking_id")
+        confirmed_moves._action_assign()
+        if pickings:
+            pickings.unlink_empty()
+
+    def _get_push_move_vals(self, move_lines):
+        """
+        Create and return a dict of values from single move in self,
+        used to create new move for push rule.
+
+        Set quantity from total of move's move lines and set link to original move.
+        """
+        self.ensure_one()
+
+        move_vals = {
+            "product_uom_qty": sum(move_lines.mapped("qty_done")),
+            "move_orig_ids": [(6, 0, self.ids)],
+        }
+        return move_vals   
+
+    @api.model
+    def _create_moves_for_push(self, push, move_lines):
+        """Create moves for a push rule to cover the quantity in move_lines"""
+        Move = self.env["stock.move"]
+
+        # Group mls by move so we can preserve move information.
+        mls_by_move = move_lines.groupby("move_id")
+        created_moves = Move.browse()
+        base_vals = {
+            "picking_type_id": push.picking_type_id.id,
+            "location_id": push.location_src_id.id,
+            "location_dest_id": push.location_id.id,
+            "picking_id": None,
+        }
+        for move, mls in mls_by_move:
+            move_vals = base_vals.copy()
+            move_vals.update(move._get_push_move_vals(mls))
+            created_moves |= move.copy(move_vals)
+        return created_moves
+    
+    def _action_done(self, cancel_backorder=False):
+        """
+        Extend _action_done to push from drop. 
+        """
+        done_moves = super(StockMove, self)._action_done(cancel_backorder=False)
+        done_moves.push_from_drop()
+        return done_moves
+
+    def _get_new_picking_values(self):
+        """
+        Extend _get_new_picking_values() to propagate the origin and parnter_id fields for next "push" picking.
+        """
+        values = super()._get_new_picking_values()
```

Comment:
nit: Sometimes we use super() and sometimes super(StockMove, self) doing the same thing, we probably only need the latter if we are mangling self or pointing it to another record set beforehand but we don't appear to be in the cases I've seen :)

Reply:
This is related to
done_moves = super(StockMove, self)._action_done(cancel_backorder=cancel_backorder)
and elsewhere possibly, could be super()._action_done(cancel_backorder=cancel_backorder)?

Reply:
This should be resolved now :)